agent/recognition/timer.py

```
@@ -0,0 +1,328 @@
+#    收尾语义得自己补全——on_error 里至少要有一个接得住的节点, 否则整条回跳栈作废、
+#    task 判失败。
+#
+# ⚠️ CustomAction **没有 detail 通道**, 功能 A 那招「状态塞进 AnalyzeResult.detail 走
+#    maafw.log 而不刷 GUI」在 Gate 上用不了。要看实时进度只能用识别器形态。
+#
+# ⚠️ 两条**尚未实机复验**的行为, 接第一条链时要盯:
+#    · 动作返回 False 后进 on_error 的**时机**——上游文档说不等 timeout, 本项目没验过;
+#    · Gate 当回边独占目标时, [JumpBack] 栈的行为是否如预期。
+#
+#    已知缺陷：
+#      计时器tag未根据ID作区分,如果中途断链再回来,动作型的计时会用到上一轮的起点，从而多算时间；
```

Ruff (0.16.3)

[warning] Comment contains ambiguous `，` (FULLWIDTH COMMA). Did you mean `,` (COMMA)?

(RUF003)